libbeat/dashboards/import_dashboards.go

```diff
@@ -448,16 +448,15 @@ func (imp Importer) ImportDir(dirType string, dir string) error {
 	// check if the directory exists
 	if _, err := os.Stat(dir); err != nil {
 		// nothing to import
-		fmt.Println("No directory", dir)
-		return nil
+		return fmt.Errorf("No directory %s", dir)
```

Comment:
Talking to myself is probably a sign of impending mental collapse but ... ok, I see the problem. The `ImportDir` method is called for different types of things which may or may not exist for a given beat.

To make things more complicated, AFAICT _none_ of the beats have a checked-in `index-pattern` dir. However, several of them have `dashboard` dirs and in fact some of them have per-module `dashboard` dirs. These latter cases are going to be missed by the test harness entirely, since it just passes `{beat}/_meta` to `import_dashboards`. I think what it _should_ be doing is finding all of the relevant subdirs and passing those one at a time to test them all.

So here's how I think this should work ...

* The `ImportDir` method should not return an error if the directory doesn't exist, but ...
  * The `ImportKibana` sub should check that the dir it is given exists and return an error if it doesn't.
* The test harness needs to be much smarter, finding all the relevant dirs for each beat and passing them one at a time.
* The test harness needs to call `libbeat/scripts/generate_index_pattern.py` for each beat so those can be imported as well.